components/tikv_util/src/lib.rs

```diff
@@ -436,16 +436,34 @@ pub fn set_panic_hook(panic_abort: bool, data_dir: &str) {
 
     let data_dir = data_dir.to_string();
     let orig_hook = panic::take_hook();
+
+    let plain_output = if let Ok(_) = env::var("TIKV_PANIC_PLAIN_OUTPUT") {
+        // Slog output needs to parsed to display multi-line strings
+        // properly. It can be easier to use the original panic hook.
+        info!("TIKV_PANIC_PLAIN_OUTPUT is set.");
+        true
+    } else {
+        false
+    };
+
     panic::set_hook(Box::new(move |info: &panic::PanicInfo<'_>| {
         use slog::Drain;
-        if slog_global::borrow_global().is_enabled(::slog::Level::Error) {
-            let msg = match info.payload().downcast_ref::<&'static str>() {
-                Some(s) => *s,
-                None => match info.payload().downcast_ref::<String>() {
-                    Some(s) => &s[..],
-                    None => "Box<Any>",
-                },
+        let msg = match info.payload().downcast_ref::<&'static str>() {
+            Some(s) => *s,
+            None => match info.payload().downcast_ref::<String>() {
+                Some(s) => &s[..],
+                None => "Box<Any>",
+            },
+        };
+
+        let use_slog =
+            if !plain_output && slog_global::borrow_global().is_enabled(::slog::Level::Error) {
```

Comment:
Actually it's quite confusing to check `Error` level here. The following usage output panic info using critical level, so here should probably check `Critical` instead. And `Critical` is always enabled, so the check is unnecessary at all. I think it's probably introduced back in the time when rust log is still used.

Reply:
That is separate from the concerns of this PR so I think that change can be done in a different PR.

Reply:
okay, I am going ahead with removing this now since it makes the PR easier to understand.